src/Microsoft.TestPlatform.Build/Tasks/VSTestForwardingTask.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.TestPlatform.Build.Trace;
+
+namespace Microsoft.TestPlatform.Build.Tasks;
+
+public class VSTestForwardingTask : Task, ITestTask
+{
+    private int _activeProcessId;
+
+    private const string DotnetExe = "dotnet";
+
+    [Required]
+    public ITaskItem? TestFileFullPath { get; set; }
+    public string? VSTestSetting { get; set; }
+    public ITaskItem[]? VSTestTestAdapterPath { get; set; }
+    public string? VSTestFramework { get; set; }
+    public string? VSTestPlatform { get; set; }
+    public string? VSTestTestCaseFilter { get; set; }
+    public string[]? VSTestLogger { get; set; }
+    public bool VSTestListTests { get; set; }
+    public string? VSTestDiag { get; set; }
+    public string[]? VSTestCLIRunSettings { get; set; }
+    [Required]
+    public ITaskItem? VSTestConsolePath { get; set; }
+    public ITaskItem? VSTestResultsDirectory { get; set; }
+    public string? VSTestVerbosity { get; set; }
+    public string[]? VSTestCollect { get; set; }
+    public bool VSTestBlame { get; set; }
+    public bool VSTestBlameCrash { get; set; }
+    public string? VSTestBlameCrashDumpType { get; set; }
+    public bool VSTestBlameCrashCollectAlways { get; set; }
+    public bool VSTestBlameHang { get; set; }
+    public string? VSTestBlameHangDumpType { get; set; }
+    public string? VSTestBlameHangTimeout { get; set; }
+    public ITaskItem? VSTestTraceDataCollectorDirectoryPath { get; set; }
+    public bool VSTestNoLogo { get; set; }
+    public string? VSTestArtifactsProcessingMode { get; set; }
+    public string? VSTestSessionCorrelationId { get; set; }
+
+    public override bool Execute()
+    {
+        var traceEnabledValue = Environment.GetEnvironmentVariable("VSTEST_BUILD_TRACE");
+        Tracing.traceEnabled = string.Equals(traceEnabledValue, "1", StringComparison.OrdinalIgnoreCase);
+
+        var debugEnabled = Environment.GetEnvironmentVariable("VSTEST_BUILD_DEBUG");
+        if (string.Equals(debugEnabled, "1", StringComparison.Ordinal))
+        {
+            Console.WriteLine("Waiting for debugger attach...");
```

Comment:
Maybe this can be part of a follow-up PR but I feel like we should not be using `Console.WriteLine`.

Reply:
The whole purpose of this class is to send output to the console, an the whole purpose of this PR is to have an alternative class that does **not** do that. So, yeah 😁